src/quest/persistence.py

```diff
@@ -19,45 +19,95 @@ def has_blob(self, key: str) -> bool: ...
 
     def delete_blob(self, key: str): ...
 
+class HistoryNode():
+    def __init__(self):
+        self.prev: HistoryNode = None
+        self.next: HistoryNode = None
+        self.item = None
 
 class PersistentHistory(History):
```

Comment:
I'd like to keep the old implementation around. Maybe call the new one `LinkedPersistentHistory` and the old one `ListPersistentHistory`. 

Then give some thought to how we might profile the performance of the two classes. I assumed that the Linked version will be more performance for prune events, but it would be nice to have some hard data for that.

Reply:
Also, do we have tests for the PersistentHistory API? If not, please add some. Test add, remove, and forward/reverse iterate; and then test the persistence using an `InMemoryBlobStorage`

Reply:
Sounds great. I have both implementations working and wrote a new test file that goes through all 4 API methods on both variations. We can talk about performance testing in class. Also, when I put `ListPersistentHistory` back in, I noticed that I could implement it using a dictionary, rather than two separate keys and items arrays. Would you prefer it to be the other way or is the dictionary implementation something you're good with? My intuition pointed out that it could be condensed, but I'm not sure it's necessarily more efficient.